tutorials/chapter1_annexB.md

```diff
@@ -0,0 +1,322 @@
+
+```C++
+_subscriber.registerListener(listenerName, [](const TestEvent &event) {
+```
+
+- 3x `destructorInvocations`, one for each `constructorInvocations` after the publishing.
+
+When the listeners finish,
+
+- 6x `destructorInvocations`: the number was incrased with the 3x `copyInvocations`.
+So, in this example:
+
+<p align="center">
+constructorInvocations + copyInvocations = destructorInvocations
+</p>
+
+## Example B2: Adding the smart loop
+
+This example shares most of the code with example B1:
+
+```diff
+- const int poolSize{0};
++ const int poolSize{8};
+```
+
+```C++
+    kpsrPublisher = eventLoop.getPublisher<TestEvent>("TestEvent",
+                                                      poolSize,
+                                                      initializerFunction,
+                                                      eventClonerFunction);
+```
+
+Any non-zero pool size when getting the publisher enables the Klepsydra smart pool: this is a memory allocation that improves the publish efficiency. See the [Smart Memory Pool](./chapter1.md#smart-memory-pool) on the main tutorial for additional details.
+
+### Sample output
+
+These are the main differences compared with example B1:
+
+```text
+$ bin/kpsr_tutorial_chapter1_annexB_2
+
+Invocations counters updating after getPublisher
+        emptyConstructorInvocations 8
+Invocations counters updating after publishing
+        constructorInvocations      3
+        destructorInvocations       3
+Invocations counters updating after slow listeners finished
+        (none)
+```
+
+**Threads.**
+
+Same 3 threads than in the previous example: main, publisher, an listener.
+
+**TestEvent constructors.**
+
+When the publisher are created:
+
+- 8x `emptyConstructorInvocations`: the smart memory pool are populated at the selected size.
+
+Just after the publishing:
+
+- same 3x `constructorInvocations` and 3x `destructorInvocations` but,
+- 0x `copyInvocations`, because listener data was aready allocated at the 8x pool.
+Go to example B5 and B6 to study a smaller pool.
+
+When the listeners finish,
+
+- `destructorInvocations` keeps the same, only the pool needs to be freed.
+
+<p align="center">
+constructorInvocations = destructorInvocations
+</p>
+
+## Example B3: Adding an initializer
+
+One step further example B2:
+
+```diff
+- const auto initializerFunction{nullptr};
++ const auto initializerFunction{[](TestEvent &event) { event.init(); }};
+```
+
+```C++
+    kpsrPublisher = eventLoop.getPublisher<TestEvent>("TestEvent",
+                                                      poolSize,
+                                                      initializerFunction,
+                                                      eventClonerFunction);
+```
+
+The `init()` function allocates the required object memory. As explained in [Smart Memory Pool](./chapter1.md#smart-memory-pool), this prevents the memory allocation on every publish event, increasing the application performance.
```

Comment:
The smart pool allocates the memory and the `initializerFunction` function can be used to perform any additional initialization step that is required by the `TestEvent` class. For example, an `std::vector` could be resized to its desired size. In this example, we have a dummy `init` function that only counts the number of invocations.